modules/context/api.go

```diff
@@ -140,3 +143,28 @@ func ReferencesGitRepo() macaron.Handler {
 		}
 	}
 }
+
+// 404 handler for APIContext
+// String will replace message, errors will be added to a slice
+func (ctx *APIContext) NotFound(objs ...interface{}) {
+	var message = "Not Found"
+	var errors []string
+	for _, obj := range objs {
+		if err, ok := obj.(error); ok {
+			errors = append(errors, err.Error())
+		} else {
+			message = obj.(string)
+		}
+	}
+
+	swagger, err := url.Parse(setting.AppURL)
```

Comment:
Without parsing the URL, the `path.Join` call was replacing the double slash in `http(s)://` with a single slash